central/networkpolicies/datastore/datastore_sac_test.go

```diff
@@ -0,0 +1,396 @@
+		s.deleteNetworkPolicy(id)
+	}
+}
+
+func (s *networkPolicySACSuite) deleteNetworkPolicy(id string) {
+	s.NoError(s.datastore.RemoveNetworkPolicy(s.testContexts[testutils.UnrestrictedReadWriteCtx], id))
+}
+
+func (s *networkPolicySACSuite) TestGetNetworkPolicy() {
+	networkPolicy := fixtures.GetScopedNetworkPolicy(uuid.NewV4().String(), testconsts.Cluster2, testconsts.NamespaceB)
+	err := s.datastore.UpsertNetworkPolicy(s.testContexts[testutils.UnrestrictedReadWriteCtx], networkPolicy)
+	s.NoError(err)
+	s.testNetworkPolicyIDs = append(s.testNetworkPolicyIDs, networkPolicy.GetId())
+	cases := map[string]struct {
+		scopeKey string
+		found    bool
+	}{
+		"global read-only can get": {
+			scopeKey: testutils.UnrestrictedReadCtx,
+			found:    true,
+		},
+		"global read-write can get": {
+			scopeKey: testutils.UnrestrictedReadWriteCtx,
+			found:    true,
+		},
+		"read-write on wrong cluster cannot get": {
+			scopeKey: testutils.Cluster1ReadWriteCtx,
+		},
+		"read-write on wrong cluster and namespace cannot get": {
+			scopeKey: testutils.Cluster1NamespaceAReadWriteCtx,
+		},
+		"read-write on wrong cluster and matching namespace cannot get": {
+			scopeKey: testutils.Cluster1NamespaceBReadWriteCtx,
+		},
+		"read-write on matching cluster can get": {
+			scopeKey: testutils.Cluster2ReadWriteCtx,
+			found:    true,
+		},
+		"read-write on matching cluster but wrong namespace cannot get": {
+			scopeKey: testutils.Cluster2NamespaceAReadWriteCtx,
+		},
+		"read-write on matching cluster and namespace can get": {
+			scopeKey: testutils.Cluster2NamespaceBReadWriteCtx,
+			found:    true,
+		},
+		"read-write on matching cluster and at least one matching namespace can get": {
+			scopeKey: testutils.Cluster2NamespacesABReadWriteCtx,
+			found:    true,
+		},
+	}
+
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := s.testContexts[c.scopeKey]
+			policy, found, err := s.datastore.GetNetworkPolicy(ctx, networkPolicy.GetId())
+			if c.found {
+				s.True(found)
+				s.NoError(err)
+				s.Equal(networkPolicy, policy)
+			} else {
+				s.False(found)
+				s.NoError(err)
+				s.Nil(policy)
+			}
+		})
+	}
+}
+
+func (s *networkPolicySACSuite) TestGetNetworkPolicies() {
+	var err error
+	networkPolicy1 := fixtures.GetScopedNetworkPolicy(uuid.NewV4().String(), testconsts.Cluster2, testconsts.NamespaceB)
+	err = s.datastore.UpsertNetworkPolicy(s.testContexts[testutils.UnrestrictedReadWriteCtx], networkPolicy1)
+	s.NoError(err)
+	s.testNetworkPolicyIDs = append(s.testNetworkPolicyIDs, networkPolicy1.GetId())
+	networkPolicy2 := fixtures.GetScopedNetworkPolicy(uuid.NewV4().String(), testconsts.Cluster3, testconsts.NamespaceB)
+	err = s.datastore.UpsertNetworkPolicy(s.testContexts[testutils.UnrestrictedReadWriteCtx], networkPolicy2)
+	s.NoError(err)
+	s.testNetworkPolicyIDs = append(s.testNetworkPolicyIDs, networkPolicy2.GetId())
+	cases := map[string]struct {
+		scopeKey      string
+		expectedErr   error
```

Comment:
We never expect an error, let's remove it + in the test func use `s.NoError(err)` instead of `s.ErrorIs()`